tests/test_sanity.py

```diff
@@ -241,3 +243,33 @@ def test_from_container_spawned_process(
         assert_collapsed(collapsed)
 
     assert b"Profiling spawned process" in container.logs()
+
+
+@pytest.mark.parametrize("in_container", [True])
+@pytest.mark.parametrize("profiler_type", ["attach-maps"])
+@pytest.mark.parametrize("runtime", ["nodejs"])
+@pytest.mark.parametrize("application_image_tag", ["without-flags"])
+def test_container_name_when_stopped(
```

Comment:
Nice test! Good job coming up with it :)

To improve - let's make it non-racey, currently it's racey and might "pass" despite the app not actually going down before the profiler did. Try to write tests that are self-guarding, i.e if you have any assumptions on the flow, then you can guard them.

One way I suggest to do it is to profile a Java app instead of nodejs. For Java, we have logs that indicate a process went down during profiling - `logger.debug(f"Profiled process {ap_proc.process.pid} exited before stopping async-profiler")`.
Your test can run the Java, begin profiling, kill the Java and ensure that this log message appears.
Also, as a best practice let's use `assert_collapsed` here to make sure the profiling of the app really worked.